### DISCUSSION TITLE
how to use MXRecordIO write multi-label？

### DISCUSSION THREAD

**Discussion by www516717402** [apache/incubator-mxnet#20171] [Q&A]
In mxnet Doc:
``` python
label = 4 # label can also be a 1-D array, for example: label = [1,2,3]
id = 2574
header = mx.recordio.IRHeader(0, label, id, 0)
with open(path, 'r') as file:
    s = file.read()
packed_s = mx.recordio.pack(header, s)
```
when my label = {'bbox':[[1,2,3,4],[5,6,7,8]],'label':[1,2].....etc}，how to storge it ?
I dont use [1,2,3,1,5,6,7,8,1....] to storge. anyone can help? thank you.

**Comment by feevos** [comment]
Hi, you can take a look in this example. I create a pickle opbject and store that in the mxnet recordio, then [on reading I unpickle it](https://discuss.mxnet.apache.org/t/writing-mxnet-ndarray-to-disk-in-recordio-format/414/2?u=feevosl). Based on this, you can put in recordio any object that can be pickled. 

Regards,
Foivos

**Reply by www516717402** [reply]
Oh,thank you for reply. However answer not my want. 
``` python
import pickle 
import mxnet as mx
record = mx.recordio.MXIndexedRecordIO(idx_path='../temp/test_train.idx',uri='../temp/test_train.rec', flag='w')

for idx in range(1000):
    # method 1
    arr = open("XXX.jpg").open()
    record.write_idx(idx,arr)
    # method 2, save fiile more large than method 1
    arr = cv2.imread("XXX.jpg")
    record.write_idx(idx,pickle.dumps(arr))
record.close()
```
